app/code/VConnect/Blog/Controller/Post/View.php

```diff
@@ -50,6 +54,18 @@ private function checkPostId(string $postId)
             );
         }
 
-        $this->postRepositoryInterface->get((int)$postId);
+        $this->post = $this->postRepositoryInterface->get((int)$postId);
+    }
+
+    /**
+     * @throws \Exception
+     */
+    private function isPostPublished()
+    {
+        if ($this->post->getPublish() === false) {
```

Comment:
Notice: probably makes sense make getPublishedPost in the repository in future. Now it is not required